src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -149,4 +140,46 @@ private function findClasses($namespace, $pattern, $excludePattern)
 
         return $classes;
     }
+
+    /**
+     * @param string $path
+     *
+     * @return string|null
+     *
+     * @link http://jarretbyrne.com/2015/06/197/
+     */
+    private function getClassFromFile($path)
```

Comment:
I really think this is a bad idea and we should stick to what I described in https://github.com/symfony/symfony/issues/22397#issuecomment-318658235
which doesn't require this magic logic.

Reply:
Yeah, I was kind of at a loss on how to implement this otherwise... Any suggestions?

Reply:
from this:
```yaml
   App\Domain\*\CommandHandler\:
        resource: ../../src/Domain/*/CommandHandler
        tags: [command_handler]
```

we should parse `App\Domain\*\CommandHandler\`:
- extract the first non variable part `App\Domain\`
- load the classes with the existing logic but *only the ones* matching `App\Domain\*\CommandHandler\` (which would have been turned into a regexp before of course, see existing logic in `AddAnnotatedClassesToCachePass::patternsToRegexps()`.)

Reply:
Thanks for the direction - I will switch my implementation.